src/components/Spot/Spot.jsx

```diff
@@ -41,13 +41,16 @@ export default function Spot({ spot, animals, organizers }) {
 			<section>
 				<h3>Organizers</h3>
 				<ul>
-					{organizers.map(organizer => {
-						let varloc = false;
-						organizer.spots.forEach(location => {
-							varloc = location.name === spot.name || location.area === spot.name;
-						});
-						return varloc && <li key={organizer.id}>{organizer.name}</li>;
-					})}
+					{organizers
```

Comment:
You made the choice to separate the data into three groups (spots, animals, organizers)  and build a reference between the entities by `name` or `id`.

There is a lot of such referencing logics scattered throughout the application code – within the code that is supposed to render content. The code is hard to read and to understand. It’s tricky to maintain and might quickly lead to bugs.

I suggest bringing this logic into a single place (like the `fetch` function from `useStore`). Do all the referencing and merging logic there. The result will be a much cleaner components.

Reply:
Please see my comment below regarding the components, that are unused at the moment.

Reply:
Coming with routing and paging...